pkg/edit/navigation.go

```diff
@@ -27,7 +27,21 @@ import (
 // Inserts the selected filename.
 
 func navInsertSelected(app cli.App) {
-	insertAtDot(app, " "+parse.Quote(navigation.SelectedName(app)))
+	fname := navigation.SelectedName(app)
+	if fname == "" {
+		// User pressed [enter] in an empty directory with nothing selected;
+		// therefore, do not insert an empty string in the buffer.
+		return
+	}
+
+	// Prepend a space to the file name only if not at the start of the line
+	// and the previous char is not a space.
+	cursor := app.CodeArea().CopyState().Buffer.Dot
```

Comment:
This operation should be a transaction inside app.CodeArea().MutateState.

Reply:
Fixed